source/server/options_impl.h

```diff
@@ -40,6 +41,9 @@ class OptionsImpl : public Server::Options, protected Logger::Loggable<Logger::I
   void setBaseId(uint64_t base_id) { base_id_ = base_id; };
   void setConcurrency(uint32_t concurrency) { concurrency_ = concurrency; }
   void setConfigPath(const std::string& config_path) { config_path_ = config_path; }
+  void setConfigProto(const envoy::config::bootstrap::v2::Bootstrap& config_proto) {
```

Comment:
Any reason to not just take this as an optional constructor arg? Then it can also be const below.

Reply:
The reason I wanted to make this a setter was because we use the argc/argv version of OptionsImpl to first initialize OptionsImpl from the command line and then later mutate it via these setter functions.

Reply:
OK sounds good.